addons/l10n_in_edi/models/account_edi_format.py

```diff
@@ -0,0 +1,442 @@
+            return super()._post_invoice_edi(invoices)
+        res = {}
+        for invoice in invoices:
+            l10n_in_edi_response_json = invoice._get_l10n_in_edi_response_json()
+            WebService = self.env["l10n.in.edi.web.service"]
+            cancel_json = {
+                "Irn": l10n_in_edi_response_json.get("Irn"),
+                "CnlRsn": invoice.l10n_in_edi_cancel_reason,
+                "CnlRem": invoice.l10n_in_edi_cancel_remarks,
+            }
+            response = WebService._cancel(invoice.company_id, cancel_json)
+            if response.get("error"):
+                error = response["error"]
+                error_codes = [e.get("code") for e in error]
+                if "1005" in error_codes:
+                    # Invalid token eror then create new token and send generate request again.
+                    # This happen when authenticate called from another odoo instance with same credentials (like. Demo/Test)
+                    authenticate_response = WebService._authenticate(invoice.company_id)
+                    if not authenticate_response.get("error"):
+                        error = []
+                        response = WebService._cancel(invoice.company_id, cancel_json)
+                        if response.get("error"):
+                            error = response["error"]
+                            error_codes = [e.get("code") for e in error]
+                if "no-credit" in error_codes:
+                    res[invoice] = {
+                        "success": False,
+                        "error": self._l10n_in_edi_get_iap_buy_credits_message(invoice.company_id),
+                        "blocking_level": "error",
+                    }
+                else:
+                    error_message = "<br/>".join(["[%s] %s" % (e.get("code"), html_escape(e.get("message"))) for e in error])
+                    res[invoice] = {
+                        "success": False,
+                        "error": error_message,
+                        "blocking_level": ("404" in error_codes) and "warning" or "error",
+                    }
+            if not response.get("error"):
+                json_dump = json.dumps(response.get("data"))
+                json_name = "%s_cancel_einvoice.json" % (invoice.name.replace("/", "_"))
+                attachment = self.env["ir.attachment"].create({
+                    "name": json_name,
+                    "raw": json_dump.encode(),
+                    "res_model": "account.move",
+                    "res_id": invoice.id,
+                    "mimetype": "application/json",
+                })
+                res[invoice] = {"success": True, "attachment": attachment}
+        return res
+
+    def _l10n_in_validate_partner(self, partner, is_company=False):
+        self.ensure_one()
+        message = []
+        if is_company and partner.country_id.code != "IN":
+            message.append(_("\n- Country should be India"))
```

Comment:
can you plz reorder the method: Invoice check configuration errors in one hand and the EDI system implementation in the other one

Reply:
@smetl this is not needed anymore because we change same in _is_required_for_invoice